test/integration/arbitrum-mainnet/Loans.fork.t.sol

```diff
@@ -463,21 +482,160 @@ contract LoansForkTest is LoansTestBase {
         assertGt(protocolFee, 0);
     }
 
-    function closeAndCheckLoan(uint loanId, uint loanAmount, uint lateFee) internal {
+    function closeAndCheckLoan(uint loanId, uint loanAmount, uint totalAmountToSwap, uint lateFee) internal {
         // Track balances before closing
         uint userCashBefore = pair.cashAsset.balanceOf(user);
         uint userUnderlyingBefore = pair.underlying.balanceOf(user);
+        uint expiration = pair.takerNFT.getPosition(loanId).expiration;
+        // Get current price from oracle
+        (uint currentPrice,) = pair.oracle.pastPriceWithFallback(uint32(expiration));
+        // Convert cash amount to expected underlying amount using oracle's conversion
+        uint expectedUnderlying = pair.oracle.convertToBaseAmount(totalAmountToSwap, currentPrice);
 
-        // no price change so underlying out should be underlying in minus slippage and minus late fees
-        uint minUnderlyingOut = underlyingAmount;
-        uint minUnderlyingOutWithSlippage = (minUnderlyingOut * (100 - slippage) / 100) - lateFee;
+        uint minUnderlyingOutWithSlippage = expectedUnderlying * (100 - slippage) / 100 - lateFee;
         uint underlyingOut = closeLoan(pair, user, loanId, minUnderlyingOutWithSlippage);
+
         assertGe(underlyingOut, minUnderlyingOutWithSlippage);
         // Verify balance changes
         assertEq(userCashBefore - pair.cashAsset.balanceOf(user), loanAmount);
         assertEq(pair.underlying.balanceOf(user) - userUnderlyingBefore, underlyingOut);
     }
 
+    // price movement settlement tests
+
+    function testSettlementPriceAboveCallStrike() public {
+        // Create provider offer & open loan
+        uint offerId = createProviderOffer(pair, callstrikeToUse, offerAmount, pair.durations[1]);
+        (uint loanId,, uint loanAmount) = openLoan(pair, user, underlyingAmount, 0.3e6, offerId);
+
+        ICollarTakerNFT.TakerPosition memory position = pair.takerNFT.getPosition(loanId);
+        skip(pair.durations[1] / 2);
+        // Move price above call strike using lib
+        PriceManipulationLib.movePriceUpPastCallStrike(
+            vm,
+            address(pair.swapperUniV3.uniV3SwapRouter()),
+            whale,
+            pair.cashAsset,
+            pair.underlying,
+            pair.oracle,
+            position.callStrikePercent,
+            POOL_FEE_TIER,
+            AMOUNT_FOR_CALL_STRIKE
+        );
+
+        skip(pair.durations[1] / 2);
```

Comment:
should probably cash duration, instead of hardcoding the index in multiple places